dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/map-to-payment-details.test.ts

```diff
@@ -0,0 +1,161 @@
+import { when } from 'jest-when';
+import {
+  CURRENCY,
+  FEE_RECORD_STATUS,
+  FeeRecordEntityMockBuilder,
+  PaymentEntityMockBuilder,
+  UtilisationReportEntity,
+  UtilisationReportEntityMockBuilder,
+} from '@ukef/dtfs2-common';
+import { FeeRecordPaymentEntityGroup } from '../../../../../types/fee-record-payment-entity-group';
+import * as helpersModule from '../../../../../helpers';
+import { TfmUsersRepo } from '../../../../../repositories/tfm-users-repo';
+import { mapToPaymentDetails } from './map-to-payment-details';
+
+describe('mapToPaymentDetails', () => {
+  const findTfmUserSpy = jest.spyOn(TfmUsersRepo, 'findOneUserById');
+  const getFeeRecordPaymentEntityGroupStatusSpy = jest.spyOn(helpersModule, 'getFeeRecordPaymentEntityGroupStatus');
+  const getFeeRecordPaymentEntityGroupReconciliationDataSpy = jest.spyOn(helpersModule, 'getFeeRecordPaymentEntityGroupReconciliationData');
+
+  beforeEach(() => {
+    findTfmUserSpy.mockRejectedValue('Some error');
+    getFeeRecordPaymentEntityGroupReconciliationDataSpy.mockResolvedValue({});
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('populates each of the groups with their respective group status', async () => {
```

Comment:
can all its in this PR start it `it('should...`. please

Reply:
Updated all of these thanks and surrounded by `describe('when...` where possible